src/casp/services/app.py

```diff
@@ -0,0 +1,184 @@
+
+
+class CloudTraceContextPlugin(Plugin):
+    key = HeaderKeys.cloud_trace_context
+
+
+class AuthorizationPlugin(Plugin):
+    key = HeaderKeys.authorization
+
+
+BASE_PLUGGINS: t.Sequence[Plugin] = (
+    # Extracts the user-agent header and makes it available to the request context.
+    plugins.UserAgentPlugin(),
+    # Extracts the x-cloud-trace-context header provided by CloudRun and makes it available to the request context.
+    CloudTraceContextPlugin(),
+    # Extracts the authorization header and makes it available to the request context.
+    AuthorizationPlugin(),
+)
+
+
+BASE_ERROR_HANDLERS: t.List[ExceptionHandlerDef] = [
+    # Default error handler for APIBaseException.  Handles most API-thrown exceptions.
+    ExceptionHandlerDef(exception=APIBaseException, handler=exception_handlers.api_base_exception_handler),
+    # Special error handler for data_mananger not found exceptions in case they are not caught.
+    ExceptionHandlerDef(exception=NotFound, handler=exception_handlers.not_found_error_handler),
+    # Error handler for token-related exceptions.
+    ExceptionHandlerDef(exception=TokenException, handler=exception_handlers.token_exception_handler),
+    # Global error handler for all other exceptions.
+    ExceptionHandlerDef(exception=Exception, handler=exception_handlers.global_error_handler),
+]
+
+
+class RequestClientMiddleware(BaseHTTPMiddleware):
+    """
+    Adds the request's client object to the request context.  This doesn't come from a
+    header which is why this can't be a plugin.
+    """
+
+    async def dispatch(self, request: Request, call_next: DispatchFunction) -> Response:
+        context[ContextKeys.client] = request.client
+        response = await call_next(request)
+        return response
+
+
+class SentryResponseDecoratorMiddleware(BaseHTTPMiddleware):
+    """
+    Adds the sentry traceid to the request response headers for support purposes.
+    """
+
+    async def dispatch(self, request: Request, call_next: DispatchFunction) -> Response:
+        response = await call_next(request)
+        # This should always be true but adding check to be safe
+        if hasattr(response, "headers"):
+            # Format is [trace_id]-[span_id]-[sampled].  We're only really interested in the trace_id.
+            trace_id = sentry_sdk.get_traceparent()
+            if trace_id:
+                trace_id = trace_id.split("-")[0]
+            response.headers[HeaderKeys.trace_id] = trace_id
+            context[ContextKeys.sentry_trace_id] = trace_id
+        return response
+
+
+class CASService(FastAPI):
+    """
+    Wrapper around FastAPI that provides some common configuration for CAS services.
+    """
+
+    def __init__(
+        self,
+        port: int,
+        plugins: t.Optional[t.Sequence[Plugin]] = None,
+        routers: t.List[RouterDef] = None,
+        exception_handlers: t.Optional[t.List[ExceptionHandlerDef]] = None,
+        sentry_application_id: str = "",
```

Comment:
What happens if `sentry_application_id` is an empty string? Wouldn't it raise an error? 

Possbile improvements: 
1. Maybe it's worth making this an optional parameter and if None then don't initialize sentry at all? 
2. Just make it a required argument (Would it work for local dev?).

Reply:
I don't think so?  I like the idea of making it optional and if it's None, we just won't initialize sentry